File: javascript/hackerrank/birthdayCakeCandles.js
// https://www.hackerrank.com/challenges/birthday-cake-candles/problem

birthdayCakeCandles = (arr) => {
    arr.sort((a, b) => a - b);
    const lastIndex = arr.length -1;
    let tallest = arr[lastIndex];
    let count = 0;		
    for (let i = lastIndex; i >= 0; i--) {
        if (arr[i] == tallest) count++;
        else break;			
    }		
    return count;
}


test("birthdayCakeCandles", () => {
    let arr = [3, 2, 1, 3];
	let result = practices.birthdayCakeCandles(arr);
	expect(result).toEqual(2);
	// threshold case 
	const n = 100000;
	arr = Array(n).fill(9999999);
	result = birthdayCakeCandles(arr);
	expect(result).toEqual(100000);
});

File: java/src/hackerrank/BirthdayCakeCandles.java
package hackerrank;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.Arrays;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

// https://www.hackerrank.com/challenges/birthday-cake-candles/problem
public class BirthdayCakeCandles {

	public static int birthdayCakeCandles(int[] arr) {
		Arrays.sort(arr);
		int lastIndex = arr.length -1;
		int tallest = arr[lastIndex];
		int count = 0;
		for (int i = lastIndex; i >= 0; i--) {
			if (arr[i] == tallest) count++;
			else break;
		}
		return count;
	}

	
	@Test
	@DisplayName("Test sample")
	void testBirthdayCakeCandles1() {
		int[] arr = {3, 2, 1, 3};
		int result = birthdayCakeCandles(arr);
		assertEquals(2, result);		
	}
	
	@Test
	@DisplayName("Test threshold case")
	void testBirthdayCakeCandles2() {
		int n = 100000;
		int[] arr = new int[n];
		for (int i=0; i<n; i++) 
			arr[i] = 9999999;
		int result = birthdayCakeCandles(arr);
		assertEquals(100000, result);		
	}
	
}
